### ISSUE TITLE
Crashes on load in node 6.2

### ISSUE THREAD

**Issue by ziggamon** [stormpath/express-stormpath#438]
```
/Users/kotliar/Code/project-backend/node_modules/cloneextend/index.js:53
   if(b.hasOwnProperty(key))
        ^

TypeError: b.hasOwnProperty is not a function
    at extend (/Users/kotliar/Code/project-backend/node_modules/cloneextend/index.js:53:9)
    at extend (/Users/kotliar/Code/project-backend/node_modules/cloneextend/index.js:66:18)
    at extend (/Users/kotliar/Code/project-backend/node_modules/cloneextend/index.js:66:18)
    at extend (/Users/kotliar/Code/project-backend/node_modules/cloneextend/index.js:66:18)
    at ExtendConfigStrategy.process (/Users/kotliar/Code/project-backend/node_modules/stormpath-config/lib/strategy/ExtendConfigStrategy.js:15:3)
    at fn (/Users/kotliar/Code/project-backend/node_modules/stormpath-config/node_modules/async/lib/async.js:746:34)
    at /Users/kotliar/Code/project-backend/node_modules/stormpath-config/node_modules/async/lib/async.js:1213:16
    at /Users/kotliar/Code/project-backend/node_modules/stormpath-config/node_modules/async/[nodemon] app crashed - waiting for file changes before starting...
```

**Comment by typerandom** [comment]
@ziggamon Could you post your configuration? I've solved some issues related to Node v6.2, but I'm unable to reproduce this one.

**Comment by ziggamon** [comment]
var stormpath_options = {  
   website: false,  
   cacheOptions : {  
       store : 'redis',  
       client : cache.client,  
       ttl: 60 * 60,  
       tti: 60 * 60,  
   },  
   expand: {  
       customData: true  
   },  
};  

On May 24 2016, at 5:12 pm, Robin &lt;notifications@github.com&gt; wrote:  

> [@ziggamon](https://github.com/ziggamon) Could you post your configuration?
I've solved some issues related to Node v6.2, but I'm unable to reproduce this
one.

>

> —  
You are receiving this because you were mentioned.  
Reply to this email directly or [view it on
GitHub](https://github.com/stormpath/express-stormpath/issues/438#issuecomment
-221302790)![](https://github.com/notifications/beacon/AAFohBAQaQa8PkdhQvdXw_Z
UUcPgeBpyks5qExUQgaJpZM4IklKS.gif)

**Comment by typerandom** [comment]
Thanks!

I'm guessing `cache.client` is a Redis client-instance? If you comment out the cacheOptions configuration, do you still have the same issue?

**Comment by ziggamon** [comment]
Yes that fixes the crash!

On May 24 2016, at 5:25 pm, Robin &lt;notifications@github.com&gt; wrote:  

> Thanks!

>

> I'm guessing `cache.client` is a Redis client-instance? If you comment out
the cacheOptions configuration, do you still have the same issue?

>

> —  
You are receiving this because you were mentioned.  
Reply to this email directly or [view it on
GitHub](https://github.com/stormpath/express-stormpath/issues/438#issuecomment
-221307597)![](https://github.com/notifications/beacon/AAFohAJAzBf7edsx3cS_4aU
jMunF_R7Lks5qExiEgaJpZM4IklKS.gif)

**Comment by typerandom** [comment]
Okay, that's great!

If you do a `console.log(cache.client)`, what does it print? What Redis client and version are you using? My guess is that `cache.client` is `undefined` or not a compatible Redis client.

**Comment by ziggamon** [comment]
REDIS: RedisClient {  
 domain: null,  
 _events:  
  { newListener: [Function],  
    error: [Function: bound ],  
    connect: [Function: bound ],  
    end: [Function: bound ] },  
 _eventsCount: 4,  
 _maxListeners: undefined,  
 address: '127.0.0.1:6379',  
 connection_options: { port: 6379, host: '127.0.0.1', family: 4 },  
 connection_id: 0,  
 connected: true,  
 ready: true,  
 should_buffer: false,  
 max_attempts: 0,  
 command_queue:  
  { [String: '']  
    '0': undefined,  
    '1': undefined,  
    _capacity: 16,  
    _length: 0,  
    _front: 2 },  
 offline_queue: { [String: ''] _capacity: 16, _length: 0, _front: 0 },  
 connect_timeout: 3600000,  
 enable_offline_queue: true,  
 retry_max_delay: null,  
 retry_timer: null,  
 retry_totaltime: 0,  
 retry_delay: 200,  
 retry_backoff: 1.7,  
 attempts: 1,  
 pub_sub_mode: false,  
 subscription_set: {},  
 monitoring: false,  
 closing: false,  
 server_info:  
  { redis_version: '3.0.7',  
    redis_git_sha1: '00000000',  
    redis_git_dirty: '0',  
    redis_build_id: 'aa27a151289c9b98',  
    redis_mode: 'standalone',  
    os: 'Darwin 15.4.0 x86_64',  
    arch_bits: '64',  
    multiplexing_api: 'kqueue',  
    gcc_version: '4.2.1',  
    process_id: '4861',  
    run_id: '654052018534b374885a759bead567045411f565',  
    tcp_port: '6379',  
    uptime_in_seconds: '498411',  
    uptime_in_days: '5',  
    hz: '10',  
    lru_clock: '4486670',  
    connected_clients: '1',  
    client_longest_output_list: '0',  
    client_biggest_input_buf: '0',  
    blocked_clients: '0',  
    used_memory: '1008416',  
    used_memory_human: '984.78K',  
    used_memory_rss: '544768',  
    used_memory_peak: '1063232',  
    used_memory_peak_human: '1.01M',  
    used_memory_lua: '36864',  
    mem_fragmentation_ratio: '0.54',  
    mem_allocator: 'libc',  
    loading: '0',  
    rdb_changes_since_last_save: '0',  
    rdb_bgsave_in_progress: '0',  
    rdb_last_save_time: '1464096370',  
    rdb_last_bgsave_status: 'ok',  
    rdb_last_bgsave_time_sec: '0',  
    rdb_current_bgsave_time_sec: '-1',  
    aof_enabled: '0',  
    aof_rewrite_in_progress: '0',  
    aof_rewrite_scheduled: '0',  
    aof_last_rewrite_time_sec: '-1',  
    aof_current_rewrite_time_sec: '-1',  
    aof_last_bgrewrite_status: 'ok',  
    aof_last_write_status: 'ok',  
    total_connections_received: '156',  
    total_commands_processed: '793',  
    instantaneous_ops_per_sec: '0',  
    total_net_input_bytes: '346246',  
    total_net_output_bytes: '418441',  
    instantaneous_input_kbps: '0.00',  
    instantaneous_output_kbps: '0.00',  
    rejected_connections: '0',  
    sync_full: '0',  
    sync_partial_ok: '0',  
    sync_partial_err: '0',  
    expired_keys: '34',  
    evicted_keys: '0',  
    keyspace_hits: '329',  
    keyspace_misses: '25',  
    pubsub_channels: '0',  
    pubsub_patterns: '0',  
    latest_fork_usec: '1207',  
    migrate_cached_sockets: '0',  
    role: 'master',  
    connected_slaves: '0',  
    master_repl_offset: '0',  
    repl_backlog_active: '0',  
    repl_backlog_size: '1048576',  
    repl_backlog_first_byte_offset: '0',  
    repl_backlog_histlen: '0',  
    used_cpu_sys: '66.09',  
    used_cpu_user: '36.73',  
    used_cpu_sys_children: '0.05',  
    used_cpu_user_children: '0.01',  
    cluster_enabled: '0',  
    versions: [ 3, 0, 7 ] },  
 auth_pass: undefined,  
 selected_db: undefined,  
 old_state: null,  
 send_anyway: false,  
 pipeline: 0,  
 times_connected: 1,  
 options:  
  { host: undefined,  
    socket_nodelay: true,  
    socket_keepalive: true,  
    return_buffers: false,  
    detect_buffers: false },  
 reply_parser:  
  JavascriptReplyParser {  
    name: 'javascript',  
    buffer: &lt;Buffer 2a 30 0d 0a&gt;,  
    offset: 5,  
    bigStrSize: 0,  
    chunksSize: 0,  
    buffers: [],  
    type: undefined,  
    protocolError: false,  
    offsetCache: 4,  
    handleReply: [Function],  
    handleNumbers: [Function],  
    returnError: [Function: returnError],  
    returnFatalError: [Function: returnFatalError],  
    returnReply: [Function: returnReply] },  
 stream:  
  Socket {  
    connecting: false,  
    _hadError: false,  
    _handle:  
     TCP {  
       bytesRead: 1899,  
       _externalStream: {},  
       fd: 18,  
       reading: true,  
       owner: [Circular],  
       onread: [Function: onread],  
       onconnection: null,  
       writeQueueSize: 0 },  
    _parent: null,  
    _host: null,  
    _readableState:  
     ReadableState {  
       objectMode: false,  
       highWaterMark: 16384,  
       buffer: [],  
       length: 0,  
       pipes: null,  
       pipesCount: 0,  
       flowing: true,  
       ended: false,  
       endEmitted: false,  
       reading: true,  
       sync: false,  
       needReadable: true,  
       emittedReadable: false,  
       readableListening: false,  
       resumeScheduled: false,  
       defaultEncoding: 'utf8',  
       ranOut: false,  
       awaitDrain: 0,  
       readingMore: false,  
       decoder: null,  
       encoding: null },  
    readable: true,  
    domain: null,  
    _events:  
     { end: [Object],  
       finish: [Function: onSocketFinish],  
       _socketEnd: [Function: onSocketEnd],  
       data: [Function],  
       error: [Function],  
       clientError: [Function],  
       close: [Object],  
       drain: [Function] },  
    _eventsCount: 8,  
    _maxListeners: undefined,  
    _writableState:  
     WritableState {  
       objectMode: false,  
       highWaterMark: 16384,  
       needDrain: false,  
       ending: false,  
       ended: false,  
       finished: false,  
       decodeStrings: false,  
       defaultEncoding: 'utf8',  
       length: 0,  
       writing: false,  
       corked: 0,  
       sync: false,  
       bufferProcessing: false,  
       onwrite: [Function],  
       writecb: null,  
       writelen: 0,  
       bufferedRequest: null,  
       lastBufferedRequest: null,  
       pendingcb: 0,  
       prefinished: false,  
       errorEmitted: false,  
       bufferedRequestCount: 0,  
       corkedRequestsFree: [Object] },  
    writable: true,  
    allowHalfOpen: false,  
    destroyed: false,  
    _bytesDispatched: 42,  
    _sockname: null,  
    _pendingData: null,  
    _pendingEncoding: '',  
    server: null,  
    _server: null,  
    read: [Function],  
    _consuming: true,  
    _idleNext: null,  
    _idlePrev: null,  
    _idleTimeout: -1 },  
 emitted_end: false,  
 uncork: [Function],  
 cork: [Function: cork] }  

On May 24 2016, at 5:35 pm, Robin &lt;notifications@github.com&gt; wrote:  

> Okay, that's great!

>

> If you do a `console.log(cache.client)`, what does it print? What Redis
client and version are you using? My guess is that `cache.client` is
`undefined` or not a compatible Redis client.

>

> —  
You are receiving this because you were mentioned.  
Reply to this email directly or [view it on
GitHub](https://github.com/stormpath/express-stormpath/issues/438#issuecomment
-221310671)![](https://github.com/notifications/beacon/AAFohJNHsIwO19m1wit2YQ5
UTOOycqjeks5qExqygaJpZM4IklKS.gif)

**Comment by ziggamon** [comment]
Any updates on this?

On May 24 2016, at 5:41 pm, Sergej Kotliar &lt;sergej.kotliar@gmail.com&gt;
wrote:  

> REDIS: RedisClient {  
 domain: null,  
 _events:  
  { newListener: [Function],  
    error: [Function: bound ],  
    connect: [Function: bound ],  
    end: [Function: bound ] },  
 _eventsCount: 4,  
 _maxListeners: undefined,  
 address: '127.0.0.1:6379',  
 connection_options: { port: 6379, host: '127.0.0.1', family: 4 },  
 connection_id: 0,  
 connected: true,  
 ready: true,  
 should_buffer: false,  
 max_attempts: 0,  
 command_queue:  
  { [String: '']  
    '0': undefined,  
    '1': undefined,  
    _capacity: 16,  
    _length: 0,  
    _front: 2 },  
 offline_queue: { [String: ''] _capacity: 16, _length: 0, _front: 0 },  
 connect_timeout: 3600000,  
 enable_offline_queue: true,  
 retry_max_delay: null,  
 retry_timer: null,  
 retry_totaltime: 0,  
 retry_delay: 200,  
 retry_backoff: 1.7,  
 attempts: 1,  
 pub_sub_mode: false,  
 subscription_set: {},  
 monitoring: false,  
 closing: false,  
 server_info:  
  { redis_version: '3.0.7',  
    redis_git_sha1: '00000000',  
    redis_git_dirty: '0',  
    redis_build_id: 'aa27a151289c9b98',  
    redis_mode: 'standalone',  
    os: 'Darwin 15.4.0 x86_64',  
    arch_bits: '64',  
    multiplexing_api: 'kqueue',  
    gcc_version: '4.2.1',  
    process_id: '4861',  
    run_id: '654052018534b374885a759bead567045411f565',  
    tcp_port: '6379',  
    uptime_in_seconds: '498411',  
    uptime_in_days: '5',  
    hz: '10',  
    lru_clock: '4486670',  
    connected_clients: '1',  
    client_longest_output_list: '0',  
    client_biggest_input_buf: '0',  
    blocked_clients: '0',  
    used_memory: '1008416',  
    used_memory_human: '984.78K',  
    used_memory_rss: '544768',  
    used_memory_peak: '1063232',  
    used_memory_peak_human: '1.01M',  
    used_memory_lua: '36864',  
    mem_fragmentation_ratio: '0.54',  
    mem_allocator: 'libc',  
    loading: '0',  
    rdb_changes_since_last_save: '0',  
    rdb_bgsave_in_progress: '0',  
    rdb_last_save_time: '1464096370',  
    rdb_last_bgsave_status: 'ok',  
    rdb_last_bgsave_time_sec: '0',  
    rdb_current_bgsave_time_sec: '-1',  
    aof_enabled: '0',  
    aof_rewrite_in_progress: '0',  
    aof_rewrite_scheduled: '0',  
    aof_last_rewrite_time_sec: '-1',  
    aof_current_rewrite_time_sec: '-1',  
    aof_last_bgrewrite_status: 'ok',  
    aof_last_write_status: 'ok',  
    total_connections_received: '156',  
    total_commands_processed: '793',  
    instantaneous_ops_per_sec: '0',  
    total_net_input_bytes: '346246',  
    total_net_output_bytes: '418441',  
    instantaneous_input_kbps: '0.00',  
    instantaneous_output_kbps: '0.00',  
    rejected_connections: '0',  
    sync_full: '0',  
    sync_partial_ok: '0',  
    sync_partial_err: '0',  
    expired_keys: '34',  
    evicted_keys: '0',  
    keyspace_hits: '329',  
    keyspace_misses: '25',  
    pubsub_channels: '0',  
    pubsub_patterns: '0',  
    latest_fork_usec: '1207',  
    migrate_cached_sockets: '0',  
    role: 'master',  
    connected_slaves: '0',  
    master_repl_offset: '0',  
    repl_backlog_active: '0',  
    repl_backlog_size: '1048576',  
    repl_backlog_first_byte_offset: '0',  
    repl_backlog_histlen: '0',  
    used_cpu_sys: '66.09',  
    used_cpu_user: '36.73',  
    used_cpu_sys_children: '0.05',  
    used_cpu_user_children: '0.01',  
    cluster_enabled: '0',  
    versions: [ 3, 0, 7 ] },  
 auth_pass: undefined,  
 selected_db: undefined,  
 old_state: null,  
 send_anyway: false,  
 pipeline: 0,  
 times_connected: 1,  
 options:  
  { host: undefined,  
    socket_nodelay: true,  
    socket_keepalive: true,  
    return_buffers: false,  
    detect_buffers: false },  
 reply_parser:  
  JavascriptReplyParser {  
    name: 'javascript',  
    buffer: &lt;Buffer 2a 30 0d 0a&gt;,  
    offset: 5,  
    bigStrSize: 0,  
    chunksSize: 0,  
    buffers: [],  
    type: undefined,  
    protocolError: false,  
    offsetCache: 4,  
    handleReply: [Function],  
    handleNumbers: [Function],  
    returnError: [Function: returnError],  
    returnFatalError: [Function: returnFatalError],  
    returnReply: [Function: returnReply] },  
 stream:  
  Socket {  
    connecting: false,  
    _hadError: false,  
    _handle:  
     TCP {  
       bytesRead: 1899,  
       _externalStream: {},  
       fd: 18,  
       reading: true,  
       owner: [Circular],  
       onread: [Function: onread],  
       onconnection: null,  
       writeQueueSize: 0 },  
    _parent: null,  
    _host: null,  
    _readableState:  
     ReadableState {  
       objectMode: false,  
       highWaterMark: 16384,  
       buffer: [],  
       length: 0,  
       pipes: null,  
       pipesCount: 0,  
       flowing: true,  
       ended: false,  
       endEmitted: false,  
       reading: true,  
       sync: false,  
       needReadable: true,  
       emittedReadable: false,  
       readableListening: false,  
       resumeScheduled: false,  
       defaultEncoding: 'utf8',  
       ranOut: false,  
       awaitDrain: 0,  
       readingMore: false,  
       decoder: null,  
       encoding: null },  
    readable: true,  
    domain: null,  
    _events:  
     { end: [Object],  
       finish: [Function: onSocketFinish],  
       _socketEnd: [Function: onSocketEnd],  
       data: [Function],  
       error: [Function],  
       clientError: [Function],  
       close: [Object],  
       drain: [Function] },  
    _eventsCount: 8,  
    _maxListeners: undefined,  
    _writableState:  
     WritableState {  
       objectMode: false,  
       highWaterMark: 16384,  
       needDrain: false,  
       ending: false,  
       ended: false,  
       finished: false,  
       decodeStrings: false,  
       defaultEncoding: 'utf8',  
       length: 0,  
       writing: false,  
       corked: 0,  
       sync: false,  
       bufferProcessing: false,  
       onwrite: [Function],  
       writecb: null,  
       writelen: 0,  
       bufferedRequest: null,  
       lastBufferedRequest: null,  
       pendingcb: 0,  
       prefinished: false,  
       errorEmitted: false,  
       bufferedRequestCount: 0,  
       corkedRequestsFree: [Object] },  
    writable: true,  
    allowHalfOpen: false,  
    destroyed: false,  
    _bytesDispatched: 42,  
    _sockname: null,  
    _pendingData: null,  
    _pendingEncoding: '',  
    server: null,  
    _server: null,  
    read: [Function],  
    _consuming: true,  
    _idleNext: null,  
    _idlePrev: null,  
    _idleTimeout: -1 },  
 emitted_end: false,  
 uncork: [Function],  
 cork: [Function: cork] }  

>

> On May 24 2016, at 5:35 pm, Robin &lt;notifications@github.com&gt; wrote:  

>

>> Okay, that's great!

>>

>> If you do a `console.log(cache.client)`, what does it print? What Redis
client and version are you using? My guess is that `cache.client` is
`undefined` or not a compatible Redis client.

>>

>> —  
You are receiving this because you were mentioned.  
Reply to this email directly or [view it on
GitHub](https://github.com/stormpath/express-stormpath/issues/438#issuecomment
-221310671)![](https://github.com/notifications/beacon/AAFohJNHsIwO19m1wit2YQ5
UTOOycqjeks5qExqygaJpZM4IklKS.gif)

**Comment by blyork** [comment]
Any update on this? I just tried updating to Node 6 and had the same issue.

**Comment by robertjd** [comment]
Hey everyone, thanks for the bump.  I'm able to reproduce the problem, and I found the cause in the underlying dependency.  I've issued a PR on our stormpath-config library to work around the problem.  I'm expediting this one for review, I hope to get this issue resolved tomorrow.

**Comment by robertjd** [comment]
This should now be fixed in the new 3.1.5 version of this library.  @blyork and @ziggamon can you upgrade to this version and let me know if it works for you in Node v6?

Thanks!

**Comment by ziggamon** [comment]
Seems to be working, thanks!
